clang/lib/Driver/ToolChains/SPIRV.cpp

```diff
@@ -70,6 +70,28 @@ void SPIRV::constructAssembleCommand(Compilation &C, const Tool &T,
                                          Exec, CmdArgs, Input, Output));
 }
 
+void SPIRV::constructLLVMLinkCommand(Compilation &C, const Tool &T,
```

Comment:
Nice improvement for SPIR-V workflows. One concern: we now have three places constructing llvm-link commands (HIP AMDGCN, HIPSPV, and the new SPIR-V path), which duplicate logic for building arguments and locating llvm-link. To keep this maintainable, could we factor the common pieces into a small helper in CommonArgs.{h,cpp}? For example, a constructLLVMLinkCommand utility that takes pre-populated LinkArgs (inputs plus static device libs) and wires up -o and the Command. Each toolchain could then keep its target specific bits (Arch, TargetID, post link steps) while sharing the common llvm-link command construction.